### PR TITLE
Create comp/core/secrets module

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

PR converts `comp/core/secrets`l to go module. In the process we create `comp/core/flare/types` module. 

### Motivation
[RFC](https://docs.google.com/document/d/1uKWa9rMNm0U-euEzEJ69KW7Id5u1E4GfYHyzC1SzYoI/edit#heading=h.wnkz5u7vx1fb)

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -233,7 +235,7 @@ require (
 	go.uber.org/atomic v1.11.0
 	go.uber.org/automaxprocs v1.5.3
 	go.uber.org/dig v1.17.0
-	go.uber.org/fx v1.18.2
+	go.uber.org/fx v1.20.1
```
Comment (hush-hush) [inline go.mod:238]: In the past we had issue with 1.20.1 breaking some dogstatsd behavrior. You might want to sync with the agent-platform team.

**Review by hush-hush — APPROVED** [review]
Looks good, just make sure to double checks my comment about dogstatsd before merging.

File: go.mod (hunk 1 above)
Comment (dineshg13) [reply go.mod]: It is not my intent to update this. It was miss. Reverted this.

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Pull request merged**